components/layout/fragment.rs

```diff
@@ -1549,66 +1444,39 @@ impl Fragment {
                 let block_flow = info.flow_ref.as_block();
                 result.union_block(&block_flow.base.intrinsic_inline_sizes)
             }
-            SpecificFragmentInfo::Image(ref mut image_fragment_info) => {
-                let mut image_inline_size = match self.style.content_inline_size() {
+            SpecificFragmentInfo::Image(_) | SpecificFragmentInfo::Canvas(_) |
```

Comment:
Can we move `is_replaced_content` to `Fragment` so we can reuse it here and everywhere else like:

```rust
_ if self.is_replaced_content() => {
   // ...
}
```

or alternatively:

```rust
if self.is_replaced_content() {
    // ....
    return ...;
}

match self.specific {
     SpecificFragmentInfo::Iframe(..) | Specifi... =>
         unreachable!("Replaced content should've been handled!"),
}
```

Since I guess we care about not forgetting to add fragment types to some parts of the layout code, and a compiler error is faster and better than a runtime error.